packages/widget-reorder/src/preview.tsx

```diff
@@ -1,11 +1,43 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { generateIdentifierFromField, copy } from './utils'
 
-export const Preview = ({ value }) => (
+export const Preview = (props) => {
+  const [data, setData] = useState<Array<object>>([])
+  const { value } = props
+
+  // Fetch initial data from localstorage
+  useEffect(() => {
+    const key = generateIdentifierFromField(props.field)
+    const metadata = localStorage.getItem(key)
+    if (metadata) setData(JSON.parse(metadata))
+  }, [])
+
+  // Reorder data when value changes
+  useEffect(() => {
+    if (data.length === 0) return
+    
+    const currentOrder = value.toJS()
+    const key: string = props.field.get('id_field')
+    const newOrder = copy({
+      from: data,
+      into: currentOrder,
+      // @ts-ignore (how can we avoid this?)
+      key
+    })
+
+    setData(newOrder)
```

Comment:
I think it'd be simpler if we store the raw data in localStorage, then in render, match each `item` with its raw value via `fieldId`.

Ooh we could also normalize the raw data before storing it (turn it from an array of object, to an object with each item's `fieldId` as key). This way in render, we can do like

```
<ul>
  {value.map(item => {
    const id = item.get(fieldId)
    const rawItem = data[id]
    return (<li>{/* do something with fieldDisplay */}</li>)
  })}
</ul>
```